powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/v3/PowerAuthServiceImpl.java

```diff
@@ -638,7 +638,7 @@ public LookupApplicationByAppKeyResponse lookupApplicationByAppKey(LookupApplica
     }
 
     @Override
-    @Transactional
+    @Transactional(rollbackFor = GenericServiceException.class)
```

Comment:
@petrdvorak can you please check this? It seems that transaction rollbacks are not handeled correctly in PowerAuth server, if I understand this change.

Reply:
Good point...